time_remaining_estimate.py

```diff
@@ -0,0 +1,58 @@
+import time
+import random
+import threading
+
+# TODO: find the best values
+GUESS_FACTOR_FAR = 600 
+GUESS_FACTOR_ALMOST = 70 
+GUESS_FACTOR_VERY_CLOSE = 10 
+
+time_estimate_sec = 0
+
+# TODO: make this a class
+
+def rand_error(spread = 10):
+    return random.randint(-spread, spread)
+
+def calculate_new_time_estimate(current_crispiness, target_crispiness):
+    """Return a time in seconds"""
+    print("generating new time estimate")
+
+    if current_crispiness is None or target_crispiness is None:
+        return 0
+
+    if current_crispiness >= target_crispiness:
+        return 0
+    elif current_crispiness == 0:
```

Comment:
wouldn't this be whenever curr_crisp<target

Reply:
Nope, this is saying if we have hit the target or passed it, show 0 seconds remaining (ie it's done). Now that I think about it, I need some tolerance on this probably...